packages/react-hooks/src/useConst.ts

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+/**
+ * Hook to initialize and return a constant value. Unlike `React.useMemo`, this is guaranteed to
+ * always return the same value (and if the initializer is a function, only call it once).
+ * This is similar to setting a private member in a class constructor.
+ *
+ * If the value should ever change based on dependencies, use `React.useMemo` instead.
+ *
+ * @param initialValue - Initial value, or function to get the initial value. Similar to `useState`,
+ * only the value/function passed in the first time this is called is respected.
+ * @returns The value. The identity of this value will always be the same.
+ *
+ * {@docCategory Hooks}
+ */
+export function useConst<T>(initialValue: T | (() => T)): T {
```

Comment:
I strongly recommend:
```ts
const valueRef = React.useRef<{ value: T; }>();
if (!valueRef.current) {
  valueRef.current = { value: typeof initialValue === 'function' ? initialValue() : initialValue };
}
return valueRef.current;
```
Reason being that `useRef` is simpler under the hood, and `useState` has some weird gotchas and edge cases. When we never need to update the value, why make React set that up?

Reply:
Good point. Fixed.